src/libraries/System.Private.CoreLib/src/System/IO/Directory.Unix.cs

```diff
@@ -20,5 +23,37 @@ private static DirectoryInfo CreateDirectoryCore(string path, UnixFileMode unixC
 
             return new DirectoryInfo(path, fullPath, isNormalized: true);
         }
+
+        private static unsafe string CreateTempSubdirectoryCore(string? prefix)
+        {
+            // mkdtemp takes a char* and overwrites the XXXXXX with six characters
+            // that'll result in a unique directory name.
+            string tempPath = Path.GetTempPath();
+            int tempPathByteCount = Encoding.UTF8.GetByteCount(tempPath);
+            int prefixByteCount = prefix is not null ? Encoding.UTF8.GetByteCount(prefix) : 0;
+            int totalByteCount = tempPathByteCount + prefixByteCount + 6 + 1;
+
+            Span<byte> path = totalByteCount <= 256 ? stackalloc byte[256].Slice(0, totalByteCount) : new byte[totalByteCount];
```

Comment:
what is the benefit of stackalloc with const-size and slice over stackalloc with variable size?

```suggestion
            Span<byte> path = totalByteCount <= 256 ? stackalloc byte[totalByteCount) : new byte[totalByteCount];
```

Reply:
See the discussion here: https://github.com/dotnet/runtime/pull/63674#discussion_r783313413

> stackalloc of a constant size is more efficient than if you pass it a variable.
*provided you are compiling with [SkipLocalsInit], which runtime is.